src/Microsoft.DotNet.Build.Tasks/PackageFiles/executor/executor.cs

```diff
@@ -0,0 +1,227 @@
+﻿using System;
+using System.IO;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+
+namespace Executor
+{
+    public class Executor
+    {
+        public Setup openFile(string filePath)
+        {
+            string jsonFile = string.Empty;
+            if (File.Exists(filePath))
+            {
+                jsonFile = File.ReadAllText(filePath);
+                Setup jsonSetup = JsonConvert.DeserializeObject<Setup>(jsonFile);
+                return jsonSetup;
+            }
+            return null;
+        }
+
+        public void usage()
+        {
+            Console.WriteLine("Executor tool");
+            Console.WriteLine("Usage: executor [config file] [command]");
+            Console.WriteLine("  Config file:      json file that contains the properties and the commands to be run. (Required)");
+            Console.WriteLine("Command: [executor command]|[config command]");
+            Console.WriteLine("  Executor command:   Executor tool commands.");
+            Console.WriteLine("         info       Provides information about the executor tool.");
+            Console.WriteLine("         help       Provides information about the content of the config file. ");
+            Console.WriteLine("  Config command:   Config file specific commands.");
+        }
+                
+        public static void Main(string[] args)
+        {
+            int argsLength = args.Length;
+            if (argsLength > 0)
+            {
+                Executor executor = new Executor();
+                Setup jsonSetup = executor.openFile(args[0]);
+                if (jsonSetup != null)
+                {
+                    //will be passed by executor.cmd/sh? we only need to now if it is in windows or non-windows.
+                    string os = "windows";
+                    //default right now.
+                    string command = "simple";
+                    bool end = false;
+                    for (int i = 1; i < argsLength && !end; i++)
+                    {
+                        switch (args[i])
+                        {
+                            case "info":
+                                executor.usage();
+                                end = true;
+                                break;
+                            case "help":
+                                Console.WriteLine("Usage: build [command]:");
```

Comment:
Yes I think build.cmd should of course remain and at least do something useful, but we can consider other options for the common entry point.

As some food for thought though today when working with msbuild as your build engine you do end up calling `msbuild /t:Test` to do your testing so using build.cmd as the entry point isn't all too much different from that. We should keep in mind that we aren't trying make this so generic that it can run everything it is just supposed to be a small abstraction around our primary build tools like msbuild and cmake for the most part. I don't think we really have a goal for example to have it ever call xunit directly for us but instead use our engine to setup the test layout and create the command and let the build engine call it.